File: app/src/main/java/ar/com/wolox/android/example/ui/home/HomeFragment.java
package ar.com.wolox.android.example.ui.home;

import android.view.View;
import android.widget.TextView;

import ar.com.wolox.android.R;
import ar.com.wolox.android.example.ui.login.LoginActivity;
import ar.com.wolox.wolmo.core.fragment.WolmoFragment;

/**
 * Home Fragment
 */
public class HomeFragment extends WolmoFragment<HomePresenter> implements HomeView {

    private TextView logoutLink;

    @Override
    public int layout() {
        return R.layout.fragment_home;
    }

    public void init() {
        logoutLink = getView().findViewById(R.id.vLogoutLink);
    }

    @Override
    public void setListeners() {
        logoutLink.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getPresenter().onLogoutLinkClick();
            }
        });
    }

    @Override
    public void logoutHome() {
        LoginActivity.start(requireContext());
    }
}

File: app/src/main/java/ar/com/wolox/android/example/ui/home/HomeFragment.kt
package ar.com.wolox.android.example.ui.home

import ar.com.wolox.android.R
import ar.com.wolox.android.example.ui.home.news.NewsFragment
import ar.com.wolox.android.example.ui.home.profile.ProfileFragment
import ar.com.wolox.wolmo.core.adapter.viewpager.SimpleFragmentPagerAdapter
import ar.com.wolox.wolmo.core.fragment.WolmoFragment
import kotlinx.android.synthetic.main.fragment_home.*
import javax.inject.Inject

class HomeFragment private constructor() : WolmoFragment<HomePresenter>(), HomeView {

    @Inject
    internal lateinit var newsFragment: NewsFragment

    @Inject
    lateinit var profileFragment: ProfileFragment

    override fun layout(): Int = R.layout.fragment_home

    override fun init() {
        vHomeViewPager.adapter = SimpleFragmentPagerAdapter(childFragmentManager).apply {
            addFragments(
                newsFragment to NEWS_TAB_TITLE,
                profileFragment to PROFILE_TAB_TITLE
            )
        }
        with(vHomeTabs) {
            setupWithViewPager(vHomeViewPager)
            getTabAt(NEWS_TAB_INDEX)!!.setIcon(R.drawable.news_tab_icon)
            getTabAt(PROFILE_TAB_INDEX)!!.setIcon(R.drawable.profile_tab_icon)
        }
    }

    companion object {
        fun newInstance() = HomeFragment()
        private const val NEWS_TAB_INDEX = 0
        private const val NEWS_TAB_TITLE = "NEWS"
        private const val PROFILE_TAB_INDEX = 1
        private const val PROFILE_TAB_TITLE = "PROFILE"
    }
}